src/lexicon/com/atproto/repo.rs

```diff
@@ -57,3 +57,34 @@ pub struct Blob {
 pub struct BlobOutput {
     pub blob: Blob,
 }
+
+/// com.atproto.repo.putRecord
+#[derive(Debug, Serialize)]
+pub struct PutRecord<T> {
+    /// The handle or DID of the repo.
+    pub repo: String,
+    /// The NSID of the record collection.
+    pub collection: String,
+    /// The key of the record.
+    /// max length of 15
+    pub rkey: String,
+    /// Validate the record?
+    /// default: true
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub validate: Option<bool>,
+    /// The record to write
+    pub record: T,
+    /// Compare and swap with the previous record by cid.
+    #[serde(rename = "swapRecord")]
+    pub swap_record: Option<String>,
```

Comment:
Note that we do not skip `swapRecord` when null, since it is `nullable` in the lexicon. Do we want some more nuanced way to express that we want to skip `swapRecord` vs. nulling it out?